common/script/ops/resetGear.js

```diff
@@ -0,0 +1,23 @@
+import _ from 'lodash';
+import content from '../content/index';
+
+module.exports = function resetGear (user) {
+  let gear = user.items.gear;
+
+  _.each(['equipped', 'costume'], function resetUserGear (type) {
+    gear[type] = {};
+    gear[type].armor = 'armor_base_0';
+    gear[type].weapon = 'weapon_warrior_0';
+    gear[type].head = 'head_base_0';
+    gear[type].shield = 'shield_base_0';
+  });
+
+  _.each(gear.owned, function resetOwnedGear (v, k) {
```

Comment:
Yea, it iterates fine. The problem is that it iterates over hidden keys. I added a condition to check those keys, and any other foreign keys.